src/test/java/com/example/rets_api/service/PropertyServiceTest.java

```diff
@@ -155,4 +158,33 @@ public void shouldDeletePropertyById_deletePropertyById() {
         verify(propertyRepositoryJPA, times(1)).delete(propertyEntity);
     }
 
+    @Test
+    public void shouldPatchProperty_patchProperty(){
+        //Creating objects
+        PropertyPatchDTO propertyPatchDTO = createPropertyPatchDTOWithBasicFields();
+        PropertyEntity propertyEntity = createPropertyEntityResponseAndID(Long.valueOf(123456789));
+
+        //Mocking calls
+        when(propertyRepositoryJPA.findById(any())).thenReturn(Optional.of(propertyEntity));
+        when(propertyRepositoryJPA.saveAndFlush(any())).thenReturn(propertyEntity);
```

Comment:
The return from findById and saveAndFlush should be different, otherwise any patch will be done inside service layer. Just create an entity to be returned in findById and another one (with different fields) to be returned in saveAndFlush :)
And the entity returned from findById must has different values from the patch, so the updatePropertyFieldsWhenChanged will be used :D